Rubberduck.CodeAnalysis/Inspections/Concrete/AssignmentNotUsedInspection.cs

```diff
@@ -108,6 +111,140 @@ private static bool IsAssignmentOfNothing(IdentifierReference reference)
+                                                .Select(lbl => (lbl.IdentifierName, lbl.Context.Start.Line));
+
+            return GotoExecutionBranchPotentiallyUsesVariable(resultCandidate, labelIdLineNumberPairs) 
+                || ResumeExecutionBranchPotentiallyUsesVariable(resultCandidate, labelIdLineNumberPairs);
+        }
+
+        private static bool GotoExecutionBranchPotentiallyUsesVariable(IdentifierReference resultCandidate, IEnumerable<(string, int)> labelIdLineNumberPairs)
+        {
+            var gotoCtxts = resultCandidate.ParentScoping.Context.GetDescendents<VBAParser.GoToStmtContext>()
+                .Where(gotoCtxt => gotoCtxt.Start.Line > resultCandidate.Context.Start.Line);
+
+            if (!gotoCtxts.Any()) { return false; }
+
+            var gotoStmt = GetFirstContextAfterLine(gotoCtxts, resultCandidate.Context.Start.Line);
+
+            if (gotoStmt == null) { return false; }
+
+            var executionBranchLine = DetermineExecutionBranchLine(gotoStmt.expression().GetText(), labelIdLineNumberPairs);
+
+            return IsPotentiallyUsedAssignment(resultCandidate, executionBranchLine);
+        }
+
+        private static bool ResumeExecutionBranchPotentiallyUsesVariable(IdentifierReference resultCandidate, IEnumerable<(string IdentifierName, int Line)> labelIdLineNumberPairs)
+        {
+            var resumeStmtCtxts = resultCandidate.ParentScoping.Context.GetDescendents<VBAParser.ResumeStmtContext>()
+                .Where(jumpCtxt => jumpCtxt.Start.Line > resultCandidate.Context.Start.Line);
+
+            if (!resumeStmtCtxts.Any()) { return false; }
+
+            var onErrorGotoStatements = resultCandidate.ParentScoping.Context.GetDescendents<VBAParser.OnErrorStmtContext>()
+                                .Where(errorStmtCtxt => !errorStmtCtxt.expression().GetText().Equals("0"))
+                                .ToDictionary(k => k.expression()?.GetText() ?? "0", v => v.Start.Line);
+
+            var errorHandlerLabelsAndLines = labelIdLineNumberPairs.Where(pair => onErrorGotoStatements.ContainsKey(pair.IdentifierName));
+            
+            //If the resultCandidate line preceeds all ErrorHandlers/Resume statements - it is not evaluated
+            if (errorHandlerLabelsAndLines.All(s => s.Line > resultCandidate.Context.Start.Line))
+            {
+                return false;
+            }
+
+            var resumeStmt = GetFirstContextAfterLine(resumeStmtCtxts, resultCandidate.Context.Start.Line);
+
+            if (resumeStmt == null) { return false; }
+
+            int? executionBranchLine = null;
+
+            var expression = resumeStmt.expression()?.GetText();
+
+            //For Resume and Resume Next, expression() is null
+            if (string.IsNullOrEmpty(expression))
+            {
+                //Get info for the errorHandlerLabel above the Resume statement
+                (string IdentifierName, int Line)? errorHandlerLabel = labelIdLineNumberPairs
```

Comment:
the `Line` isn't used subsequently... it might be better to perform a `Select` to remove it?

Reply:
Yes...refactored